horton/io/vasp.py

```diff
@@ -39,51 +39,62 @@ def _unravel_counter(counter, shape):
     return result
 
 
-def _load_vasp_header(f, nskip):
+def _load_vasp_header(f):
     '''Load the cell and atoms from a VASP file
+       File specification provided here:
+        http://cms.mpi.univie.ac.at/vasp/guide/node59.html
 
        **Arguments:**
 
        f
             An open file object
 
-       nskip
-            The number of lines to skip after the line with elements
-
        **Returns:** ``title``, ``cell``, ``numbers``, ``coordinates``
     '''
-    # reat the title
+    # read the title
     title = f.next().strip()
-    f.next()
+    # read the universal scaling factor
+    scaling = float(f.next().strip())
 
-    # read cell parameters in angstrom. each row is one cell vector
+    # read cell parameters in angstrom, without the universal scaling factor.
+    # each row is one cell vector
     rvecs = []
     for i in xrange(3):
         rvecs.append([float(w) for w in f.next().split()])
-    rvecs = np.array(rvecs)*angstrom
+    rvecs = np.array(rvecs)*angstrom*scaling
 
     # Convert to cell object
     cell = Cell(rvecs)
 
+    # note that in older VASP version the following line might be absent
     vasp_numbers = [periodic[w].number for w in f.next().split()]
     vasp_counts = [int(w) for w in f.next().split()]
     numbers = []
     for n, c in zip(vasp_numbers, vasp_counts):
         numbers.extend([n]*c)
     numbers = np.array(numbers)
 
-    # skip some lines
-    for i in xrange(nskip):
-        f.next()
-    assert f.next().startswith('Direct')
-
-    # read the fractional coordinates and convert to Cartesian
+    line = f.next()
+    # the 7th line can optionally indicate selective dynamics
+    if line[0].lower() in ['s']:
+        line = f.next()
+    # parse direct/cartesian switch
+    if line[0].lower() in ['c','k']:
```

Comment:
Could you use `cartesian = line[0].lower() in ['c', 'k']`, which is nicer. Current code fails on one of our PyLint tests.